views/home/main.js

```diff
@@ -152,7 +153,7 @@ function turn_all_switches_off() {
   for (let i = 0; i < toggle_switch_elements.length; i++) {
     toggle_switch_elements[i].checked = false;
   }
-  num_toggled_elements = 0;
+  toggles = [];
```

Comment:
The idea behind this (I told leah to do this), would be that it's a lot faster to check if there's an element toggled if you keep a number of which ones are toggled as opposed to keeping a list where'd you have to look through all of them.

Reply:
This was the cause of the bug; the toggles array was not added to when the length was incremented or vice versa. JS lists maintain their own lengths so we should take advantage of that and not try and remember to change two variables for each insertion and deletion.

Reply:
This isn't a hill I'm willing to die on, but shouldn't this work? If you simply add the toggle to the array, the toggle's value could be true or false. The benefit of this approach would be that while you have to keep track of the number of elements, just looking at that number would tell you if there's an element currently toggled.

In hindsight though, I am realizing that this is a lot of additional work and the best approach would be to just clear the states like you mentioned below and setting all of the toggles to "o"

Reply:
Array.length is the value that we can look at to see if any toggles are on. It is a constant-time operation since the value is just a property that gets updated whenever an element is added to or removed from the array, so it eliminates the need for us to keep track outside of it. It never has to traverse the entire array to get the length.

Reply:
Also the list is required because we need to send the IDs of the toggles to the backend to generate the UUID, so we need to know which toggles are on, not just the number of toggles that are on. Sorry if I wasn't clear about that.

Reply:
> Sorry if I wasn't clear about that.
You're good, the original approach I suggested is overkill even if it did work so we can just scrap it. I tried to get rid of my comment before you saw it but it looks like I failed, my bad!